File: app/src/main/java/org/schulcloud/mobile/data/model/Course.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmList;
import io.realm.RealmModel;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
public class Course implements RealmModel {
    @PrimaryKey
    public String _id;
    public String schoolId;
    public String name;
    public String description;
    public String color;
    public RealmList<Times> times;
    public String startDate;
    public String untilDate;
    public Boolean gradeSystem;
    public RealmList<User> substitutionIds;
    public RealmList<User> teacherIds;
    public RealmList<User> userIds;
}

File: app/src/main/java/org/schulcloud/mobile/models/course/Course.kt
package org.schulcloud.mobile.models.course

import com.google.gson.annotations.SerializedName
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.schulcloud.mobile.models.base.HasId
import org.schulcloud.mobile.models.user.User

open class Course : RealmObject(), HasId {

    @PrimaryKey
    @SerializedName("_id")
    override var id: String = ""

    var schoolId: String? = null
    var name: String? = null
    var description: String? = null
    var color: String? = null

    @SerializedName("teacherIds")
    var teachers: RealmList<User>? = null

}
